ksmt-symfpu/src/main/kotlin/org/ksmt/symfpu/SymFPUModel.kt

```diff
@@ -0,0 +1,160 @@
+package org.ksmt.symfpu
+
+import org.ksmt.KContext
+import org.ksmt.decl.KDecl
+import org.ksmt.expr.KConst
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.KUninterpretedSortValue
+import org.ksmt.solver.KModel
+import org.ksmt.solver.model.KModelEvaluator
+import org.ksmt.solver.model.KModelImpl
+import org.ksmt.sort.KArraySortBase
+import org.ksmt.sort.KFpSort
+import org.ksmt.sort.KSort
+import org.ksmt.sort.KUninterpretedSort
+import org.ksmt.symfpu.ArraysTransform.Companion.mkAnyArrayLambda
+import org.ksmt.symfpu.ArraysTransform.Companion.mkAnyArraySelect
+import org.ksmt.utils.cast
+import org.ksmt.utils.uncheckedCast
+
+class SymFPUModel(private val kModel: KModel, val ctx: KContext, val transformer: FpToBvTransformer) : KModel {
+    override val declarations: Set<KDecl<*>>
+        get() = kModel.declarations + transformer.arraysTransform.mapFpToBvDeclImpl.keys
```

Comment:
`kModel.declarations` contains our internal `bv` declarations thar are used as a replacement for the `fp` decls. We must hide such declarations.